spacy/pipeline/entity_linker.py

```diff
@@ -455,10 +469,19 @@ def predict(self, docs: Iterable[Doc]) -> List[str]:
                             if sims.shape != prior_probs.shape:
                                 raise ValueError(Errors.E161)
                             scores = prior_probs + sims - (prior_probs * sims)
-                        # TODO: thresholding
-                        best_index = scores.argmax().item()
-                        best_candidate = candidates[best_index]
-                        final_kb_ids.append(best_candidate.entity_)
+                        meets_abst_threshold = (
+                            xp.where(scores >= self.abstention_threshold)[0]
+                            if self.abstention_threshold is not None
+                            else xp.arange(len(scores))
+                        )
+                        best_candidate_entity_ = (
+                            candidates[
+                                meets_abst_threshold[scores.argmax().item()]
+                            ].entity_
+                            if meets_abst_threshold.size
+                            else EntityLinker.NIL
+                        )
+                        final_kb_ids.append(best_candidate_entity_)
```

Comment:
It feels to me like the rewrite is unnecessarily complex. You're comparing every score to the threshold and still taking a max, though we might as well just take the max first (as in the original code) and only do one final check against the `max_score` and either define & append `best_candidate.entity_`, or append `NIL` if the highest score is below the threshold.

Reply:
That's absolutely correct :sweat_smile: